ui/app/models/aws-credential.js

```diff
@@ -3,21 +3,26 @@ import Ember from 'ember';
 import { expandAttributeMeta } from 'vault/utils/field-to-attrs';
 const { attr } = DS;
 const { computed } = Ember;
-const CREATE_FIELDS = ['ttl'];
+const CREATE_FIELDS = ['ttl', 'roleArn'];
 
 const DISPLAY_FIELDS = ['accessKey', 'secretKey', 'securityToken', 'leaseId', 'renewable', 'leaseDuration'];
 export default DS.Model.extend({
+  helpText:
+    'For Vault roles that have a credential type of iam_user, these attributes are optional and you may simply submit the form.',
   role: attr('object', {
     readOnly: true,
   }),
 
-  withSTS: attr('boolean', {
-    readOnly: true,
+  roleArn: attr('string', {
+    label: 'Role ARN',
+    helpText:
+      'The ARN of the role to assume if credential_type on the Vault role is assumed_role. Optional if the role has a single role ARN; required otherwise.',
   }),
-
   ttl: attr({
     editType: 'ttl',
-    defaultValue: '1h',
```

Comment:
I think this might be a bug. When I pull up the page to generate a new credential, it offers me a default of 30 minutes, which I think comes from https://github.com/hashicorp/vault/blob/c9a4fdf143040d856a1610df41945d1949e69417/ui/app/components/ttl-picker.js#L13-L14

However, the default for the backend is actually 60m (3600s): https://github.com/hashicorp/vault/blob/c9a4fdf143040d856a1610df41945d1949e69417/builtin/logical/aws/path_user.go#L32

As long as the UI passes the default value back explicitly, this isn't a problem. However, when the TTL is left at the default, the UI doesn't pass any data back to the Vault server, and as a result, a credential gets generated with a TTL of 3600s, which isn't what users would have thought they were requesting.

Reply:
See #5203 which should fix this.